projects/addon-doc/src/interfaces/page.ts

```diff
@@ -11,3 +11,23 @@ export interface TuiDocPage extends TuiDocPageBase {
 export interface TuiDocPageGroup extends TuiDocPageBase {
     readonly subPages: ReadonlyArray<TuiDocPage>;
 }
+
+export const EXAMPLE_PRIMARY_FILE_NAME = {
+    TS: 'TypeScript',
+    LESS: 'LESS',
+    HTML: 'HTML',
+} as const;
+
+export type RawLoaderContent = Promise<{default: string}>;
+
+export type TuiDocCodeValue = RawLoaderContent | string | null | undefined;
+
+export interface TuiDocExample extends Record<string, TuiDocCodeValue> {
+    [EXAMPLE_PRIMARY_FILE_NAME.TS]?: TuiDocCodeValue;
+    [EXAMPLE_PRIMARY_FILE_NAME.HTML]?: TuiDocCodeValue;
+    [EXAMPLE_PRIMARY_FILE_NAME.LESS]?: TuiDocCodeValue;
+}
+
+export type TuiDocCodeProcessedValue = string | null | undefined;
```

Comment:
why do we need so many `falsy` values ?
it produces a lot of optional chaining in code :(

Maybe it better to keep only `string` (and return empty string where it is necessary to indicate `falsy` value) ?